addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComHandler.java

```diff
@@ -0,0 +1,322 @@
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.rfxcom.RFXComValueSelector;
+import org.openhab.binding.rfxcom.RFXComBindingConstants;
+import org.openhab.binding.rfxcom.internal.DeviceMessageListener;
+import org.openhab.binding.rfxcom.internal.RFXComException;
+import org.openhab.binding.rfxcom.internal.messages.RFXComBaseMessage;
+import org.openhab.binding.rfxcom.internal.messages.RFXComMessage;
+import org.openhab.binding.rfxcom.internal.messages.RFXComMessageFactory;
+import org.openhab.binding.rfxcom.internal.messages.RFXComBaseMessage.PacketType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RFXComHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ * 
+ * @author Pauli Anttila - Initial contribution
+ */
+public class RFXComHandler extends BaseThingHandler implements DeviceMessageListener {
+
+    private Logger logger = LoggerFactory.getLogger(RFXComHandler.class);
+
+	private int refresh = 10;
+	ScheduledFuture<?> refreshJob;
+	private RFXComBridgeHandler bridgeHandler;
+	
+	private String deviceId = null;
+	private String subType = null;
+	
+	public RFXComHandler(Thing thing) {
+		super(thing);
+	}
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		logger.debug("Received channel: {}, command: {} (this={})", channelUID, command, this);
+
+		getBridgeHandler();
+
+		if (bridgeHandler != null) {
+
+			try {
+				PacketType packetType = RFXComMessageFactory
+						.convertPacketType(channelUID.getThingTypeId()
+								.toUpperCase());
+
+				RFXComMessage msg = RFXComMessageFactory
+						.getMessageInterface(packetType);
+
+				List<RFXComValueSelector> supportedValueSelectors = msg
+						.getSupportedOutputValueSelectors();
+
+				RFXComValueSelector valSelector = RFXComValueSelector
+						.getValueSelector(channelUID.getId());
+
+				if (supportedValueSelectors.contains(valSelector)) {
+					msg.setSubType(msg.convertSubType(subType));
+					msg.setDeviceId(deviceId);
+					msg.convertFromState(valSelector, command);
+					
+					bridgeHandler.sendMessage(msg);
+				} else {
+					logger.warn(
+							"RFXCOM doesn't support tranmitting for channel '{}'",
+							channelUID.getId());
+				}
+
+			} catch (RFXComException e) {
+				logger.error("Transmitting error", e);
+			}
+		}
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void initialize() {
+
+		logger.debug("Initialized RFXCOM device handler for {}.", getThing()
+				.getUID());
+
+		Configuration config = getThing().getConfiguration();
+		deviceId = (String) config.get(RFXComBindingConstants.DEVICE_ID);
+		subType = config.get(RFXComBindingConstants.SUB_TYPE).toString();
+		
+		logger.debug("deviceId={}, subType={}", deviceId, subType);
+		
+		getBridgeHandler();
```

Comment:
all the following code of this method is not necessary anymore, see bridgeHandlerInitialized() method